pkg/cmd/util/clientcmd/factory.go

```diff
@@ -714,6 +731,10 @@ func (f *Factory) PodForResource(resource string, timeout time.Duration) (string
 	case api.Resource("pods"):
 		return name, nil
 	case api.Resource("replicationcontrollers"):
```

Comment:
we could hit all these k8s resources and fail against a kubernetes server, right?